core/src/main/java/org/infinispan/persistence/async/AsyncCacheWriter.java

```diff
@@ -201,14 +207,18 @@ public void write(MarshallableEntry entry) {
    }
 
    @Override
-   public void writeBatch(Publisher publisher) {
-      putAll(
-            Flowable.fromPublisher((Publisher<MarshallableEntry>) publisher)
-                  .map(me -> new Store(me.getKey(), me))
-                  .cast(Modification.class)
-                  .toList()
-                  .blockingGet()
-      );
+   public CompletionStage<Void> writeBatch(Publisher publisher) {
+      CompletableFuture<Void> future = new CompletableFuture<>();
+      Flowable.fromPublisher((Publisher<MarshallableEntry>) publisher)
+            .subscribeOn(scheduler)
```

Comment:
I would say we don't need the scheduler configured here. The only reason the scheduler should be used in the store itself is to add additional parallelism that you can't obtain by methods on the provided Publisher. For example we could do parallelism to do each batch insertion, but I don't want to do that yet.